ouroboros-network-api/src/Ouroboros/Network/AnchoredSeq.hs

```diff
@@ -295,6 +296,26 @@ splitAt i (AnchoredSeq a ft) = case FT.split (\v -> measureSize v > i) ft of
      let before' = AnchoredSeq a before
      in (before', AnchoredSeq (headAnchor before') after)
 
+-- | \( O(\log(\min(i,n-i)) \). Split the 'AnchoredSeq' at a given
+--  measure.
+--
+-- POSTCONDITION: @(before, after) = splitAtMeasure v s@, then:
+--
+-- > anchor      before == anchor s
+-- > headAnchor  before == anchor after
+-- > headAnchor  after  == headAnchor s
+-- > [before] == filter ((< v) . getElementMeasure . MeasuredWith) s
+-- > [after] == filter ((v <=) . getElementMeasure . MeasuredWith) s
```

Comment:
It would nice and relatively straightforward to add a property test for this to `Test.AnchoredFragment`.

Also, `getElementMeasure` and `MeasuredWith` are internal. Not a big deal per se; when testing, we can extract the slot number directly. Might be another small advantage of adding `splitAtSlot` as its laws could be tested unmodified.

Reply:
Added `splitAtSlot` and the tests.